### PR TITLE
feat: added search functionality

### PR DESCRIPTION
**Pull request by dev-ashishsethi**
image
Files to look for:
Search.jsx
SearchContext.jsx

### CODE REVIEW

File: src/context/SearchContext.jsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+export function SearchProvider({ children }) {
+  const [searchTerm, setSearchTerm] = useState("");
+  const [searchResult, setSearchResult] = useState([]);
+
+  useEffect(() => {
+    (async () => {
+      const response = await axios.get("/api/products");
+      console.log("from search context result", response.data.products);
+      console.log("search term in context", searchTerm.toLowerCase());
```
Comment (m26unkwn) [inline src/context/SearchContext.jsx:14]: Remove this whole console.log from your production code.

**Review by m26unkwn — APPROVED** [review]
Everything is good to go just look for the comments.

1. Abstract all the functionality from jsx

File: src/context/SearchContext.jsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+      setSearchResult(
+        response.data.products.filter((products) =>
+          products.title.toLowerCase().includes(searchTerm.toLowerCase())
+        )
+      );
+    })();
+  }, [searchTerm]);
+
+  console.log(searchResult);
```
Comment (PuneetP16) [inline src/context/SearchContext.jsx:24]: remove console  logs before commits

File: src/pages/Search/Search.jsx [hunk 3]
```diff
@@ -0,0 +1,142 @@
+import "./Search.css";
+import { useState } from "react";
+import { useCart } from "../../context/CartContext";
+import { useSearch } from "../../context/SearchContext";
+import { useWishList } from "../../context/WishlistContext";
```
Comment (PuneetP16) [inline src/pages/Search/Search.jsx:5]: Suggestion: you could add all the context in one index.js file and import here in one line

File: src/pages/Search/Search.jsx [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+        });
+        setCart(response.data.cart);
+      } catch (error) {
+        console.error(error);
+      }
+    })();
+  };
+  const addToWishList = (items) => {
+    const token = localStorage.getItem("login");
+    console.log("items", items);
```
Comment (PuneetP16) [inline src/pages/Search/Search.jsx:44]: here too

**Review by PuneetP16 — APPROVED** [review]
Appreciation: Your UI and search functionality is working good. Check for the comments, else good to merge.

**Pull request merged**